### ISSUE TITLE
use scenery 'cachedPaints' feature to improve performance

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/sun#150]
Related to https://github.com/phetsims/sun/issues/147.  Scenery has a new 'cachedPaints' feature that caches gradients, patterns, etc.  See scenery.Paintable.  This should be used in sun to improve performance.

**Comment by pixelzoom** [comment]
Marking this as high priority, since all sims will take a performance hit.

**Comment by pixelzoom** [comment]
Assigning to @jbphet since he was the primary architect of sun buttons, and was involved in using the scenery 0.1 "fillKept" feature.

**Comment by jonathanolson** [comment]
Added cachedPaints where appropriate (basically everywhere we are switching between fills/gradients).

Will wait for @jbphet to review.

**Comment by jbphet** [comment]
The commits look good based on my understanding of how the cached paints feature works, which in turn is based on what I see in the Scenery documentation here: http://phetsims.github.io/scenery/doc/#paintable-cachedPaints and a look at the Scenery code.  I removed one cached gradient that appeared to be redundant (see commit above) and did some other minor code cleanup.  I tested on Win7 using Chrome, Firefox, and IE11, and on and iPad 2 running iOS 7.0.4 using the sun demo (sun_en.html).  I took heap snapshots in Chrome and verified that the number of allocated LinearGradients and RadialGradients was not increasing when the button base color was changed.  All in all, this appears to have been fully addressed.  Closing.
